### DISCUSSION TITLE
How to get request size on server side from inside tower layer?

### DISCUSSION THREAD

**Discussion by gmossessian** [hyperium/tonic#1294] [Q&A]
Is it possible to read the size in bytes of a `tonic::codegen::http::Request<T>`, or is the size only available once the inner `prost::Message` is consumed?

**Comment by LucioFranco** [comment]
Yeah, I don't think there is a way to peek at the header before its consumed by prost. You could always write a http body wrapper that tracks the bytes and reports them as it gets read from the buffer.
